sockeye/utils.py

```diff
@@ -30,13 +30,57 @@
 import mxnet as mx
 import numpy as np
 
+from sockeye import __version__
+
 logger = logging.getLogger(__name__)
 
 
 class SockeyeError(Exception):
     pass
 
 
+def check_version(version: str):
+    """
+    Checks given version against code version and determines compatibility.
+    Throws if versions are incompatible.
+
+    :param version: Given version.
+    """
+    code_version = parse_version(__version__)
+    given_version = parse_version(version)
+    check_condition(code_version[0] == given_version[0],
+                    "Given release version (%s) does not match release code version (%s)" % (version, __version__))
+    check_condition(code_version[1] == given_version[1],
+                    "Given major version (%s) does not match major code version (%s)" % (version, __version__))
+    if code_version[2] != given_version[2]:
+        logger.warning("Given minor version (%s) does not match code version (%s)", version, __version__)
```

Comment:
ah yeah, that's what I meant. I just mean that a warning 'A did not B' sounds concerning, while in reality a mismatch shouldn't be concerning.

We're already logging the current Sockeye version and with the logging statement of the model Sockeye version in inference.py I think it would be ok to leave out this warning or info message here.